tests/Microsoft.Identity.Test.Unit/PublicApiTests/ConfidentialClientApplicationTests.cs

```diff
@@ -1672,6 +1672,48 @@ public async Task ValidateGetAccountAsyncWithNullEmptyAccountIdAsync(string acco
                 Assert.IsNull(acc);
             }
         }
+
+        [TestMethod]
+        public async Task ValidateAppTokenProviderAsync()
```

Comment:
We should have a test where the expiration is 1 second to see if expiration really works.

Reply:
Also another call with different scopes (assert Idp source + 2 tokens in cache).